webhook_features/src/features/summary_comment.rs

```diff
@@ -0,0 +1,633 @@
+        if let Some(cmt_id) = &cmt.id {
+            if cmt_id.to_string() == comment.id {
+                continue;
+            }
+        }
+
+        let acks = parse_acks_in_comment(&comment.body); // A single comment can contain multiple acks, e.g 'Concept ACK and Code Review ACK'
+
+        for ack_commit in acks {
+            let ack_type = ack_commit.ack_type.clone();
+            let commit = ack_commit.commit.clone();
+
+            if commit.is_none() && ack_type.requires_commit_hash() {
+                // If the ack type requires a commit hash, but the comment does not contain one, skip it
+                continue;
+            }
+
+            if ack_per_user.contains_key(&comment.user) // If the user already has an ack for this commit
+                && should_skip_ack(ack_commit.clone(), ack_per_user[&comment.user].clone())
+            // Maybe the ack is a duplicate, need to check for Stale acks
+            {
+                continue;
+            }
+
+            let head = commit.is_some() && commit.as_ref().map_or(false, |c| c.0 == pr.head.sha); // Check if the commit is the head commit of the PR
+
+            parsed_acks.push(Review {
+                user: comment.user.clone(),
+                ack_type,
+                commit: commit.map(|c| (c, head)),
+                url: comment.url.clone(),
+            });
+
+            ack_per_user
+                .entry(comment.user.clone())
+                .or_insert_with(|| vec![ack_commit.clone()]);
+        }
+    }
+
+    let comment = summary_comment_template(false, Some(parsed_acks));
+    util::update_metadata_comment(
+        &ctx.octocrab.issues(&repo.owner, &repo.name),
+        cmt,
+        &comment,
+        util::IdComment::SecReviews,
+        false).await?;
+    Ok(())
+}
+
+#[derive(Clone, Copy, Debug, Hash, PartialEq, Eq)]
+enum AckType {
+    ACK,
+    NACK,
+    CodeReviewACK,
+    ConceptACK,
+    ApproachACK,
+    ApproachNACK,
+
+    StaleACK, // ACK, but the commit is not the head of the PR anymore
+}
+
+impl AckType {
+    fn requires_commit_hash(&self) -> bool {
+        match self {
+            AckType::ACK | AckType::CodeReviewACK => true,
+            _ => false,
+        }
+    }
+
+    fn to_string(&self) -> &str {
+        match self {
+            AckType::ACK => "ACK",
+            AckType::NACK => "NACK",
+            AckType::CodeReviewACK => "Code-Review ACK",
+            AckType::ConceptACK => "Concept ACK",
+            AckType::ApproachACK => "Approach ACK",
+            AckType::ApproachNACK => "Approach NACK",
+            AckType::StaleACK => "Stale ACK",
+        }
+    }
+}
+
+macro_rules! multi_vec {
+    ($([$($key:literal),+] => $value:expr);*) => {
+        vec![
+            $($(($key, $value)),*),*
+        ]
+    };
+}
+
+lazy_static! {
+    static ref ACK_PATTERNS: Vec<(&'static str, AckType)> = multi_vec![
+        ["code review ack", "cr ack", "cr-ack", "crack"] => AckType::CodeReviewACK;
```

Comment:
Why not map this to ::ACK? Sure it is possible that someone did *not* review the code (maybe only did testing in the GUI or so), but then this should be mentioned in the review comment, see https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#code-review